### ISSUE TITLE
test case 3 fail

### ISSUE THREAD

**Issue by haoyanggggg** [nus-cs2030/2223-s2#614]
# Summary

passed 2 test cases but failed 3rd one

## Description:

can someone give me some insights what am i doing wrongly? currently hv no rest class, just implementing getresttime in done class to update nextavailtime for that specific server. also i passed the first 3 test cases on the lab 5 page (no self checkout counters), so im not sure why i failed halfway through the private test cases

## Screenshots

![image](https://user-images.githubusercontent.com/124251722/227754137-da6024fe-41f8-41c3-9955-c30316f3696d.png)

**Comment by reinatng10** [comment]
Hi Haoyang, not very sure where went wrong but what i did was i updated my serverTime to serverTime + restTime in my DoneEvent class. Then i return new Pair<Event, ImList<Server>>(this, serverList).

In my simulator class, for every doneEvent i take out i do this: serverList = nextEvent.second() 
to update my serverList.

**Comment by haoyanggggg** [comment]
> Hi Haoyang, not very sure where went wrong but what i did was i updated my serverTime to serverTime + restTime in my DoneEvent class. Then i return new Pair<Event, ImList>(this, serverList).
> 
> In my simulator class, for every doneEvent i take out i do this: serverList = nextEvent.second() to update my serverList.

yup that's similar to what im doing, except im doing pair<pq<event>, imlist<server> (pq, serverlist). in simulator i also update my serverlist correctly by using second(). did u change anything else in other event classes other than doneevent?

**Comment by trickypepe** [comment]
Not very sure what went wrong but from what I can see it looks like server 1 skipped his break when there are people in queue (10 immediately served by server 1 after server 1 is done with 9). Maybe you can create a test case to try this (as for the correct output you could probably check with your TA or friend who finished).

**Comment by Sangobionn** [comment]
perhaps you can try checking if the server is indeed taking a break (idk how you check for your implementation - perhaps you have a boolean attribute in Server class that is checking if free). Make sure that server is not free when taking a break.

personally for me I:
1. added a break event class 
2. edited my ArrivalEvent and BeforeServe event to update server list before running the event (to make servers that are not free && time of event >= server's end time free now) this is to ensure servers can serve after the break is up. This ensures customers that just arrive and  queueing alr can be served by now-free-server.
I am using the Pair<Event, serverlist> in my Events

**Comment by stxe09** [comment]
where are you calling getRestTime()? if you call it in Serve, you will get issues like this

**Comment by youthfrost** [comment]
You probably didn't update your servers at one of the events, that's why your getRestTime is called more than it's suppose to be, resulting in discrepancy on the timestamp
